employee/src/main/java/org/esteco/anniversary/ApplicationConfig.java

```diff
@@ -1,10 +1,73 @@
-package org.esteco.anniversary;
-
-import org.springframework.context.annotation.ComponentScan;
-import org.springframework.context.annotation.Configuration;
-
-@Configuration
-@ComponentScan(basePackages = "org.esteco.anniversary")
-public class ApplicationConfig {
-
-}
+package org.esteco.anniversary;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.dao.annotation.PersistenceExceptionTranslationPostProcessor;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.JpaVendorAdapter;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.Database;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+import java.util.Properties;
+
+
+@Configuration
+@ComponentScan(basePackages = "org.esteco.anniversary")
+@EnableTransactionManagement
+public class ApplicationConfig {
+
+    @Bean(name="dataSource")
+    public DataSource dataSource(){
+        BasicDataSource dataSource = new BasicDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
+        dataSource.setUrl("jdbc:mysql://localhost:3306/esteco");
+        dataSource.setUsername("root");
+        dataSource.setPassword("ath@678AT");
+        return dataSource;
+    }
+
+    @Bean
+    public JpaVendorAdapter jpaVendorAdaptor(){
+        HibernateJpaVendorAdapter adapter = new HibernateJpaVendorAdapter();
+        adapter.setDatabasePlatform("org.hibernate.dialect.MySQL5Dialect");
+        adapter.setShowSql(true);
+        adapter.setDatabase(Database.MYSQL);
+        adapter.setGenerateDdl(true);
+        return adapter;
+    }
+
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory(
+            DataSource dataSource, JpaVendorAdapter jpaVendorAdapter){
+
+        Properties prop = new Properties();
+        prop.setProperty("hibernate.format_sql",String.valueOf(true));
+
+        LocalContainerEntityManagerFactoryBean emf = new LocalContainerEntityManagerFactoryBean();
+        emf.setDataSource(dataSource);
+        emf.setJpaVendorAdapter(jpaVendorAdapter);
+        emf.setJpaProperties(prop);
+        emf.setPackagesToScan("org.esteco.anniversary");
+
+        return emf;
+    }
+
```

Comment:
Okay.  Because, we are using sprint orm we will not need JPA dependency.

Reply:
Yep!